### ISSUE TITLE
Missing ATX headings in markdown pages

### ISSUE THREAD

**Issue by jschleus** [doxygen/doxygen#6925]
If Doxygen renders markdown pages containing ATX headings (using hashes) into HTML in many case it doesn't output the <Hn> HTML heading lines.

My impression is that Doxygen  uses the first heading line for generating the "headertitle" of the resulting HTML page but "forgets" respectively "ignores" that found heading line while the further parsing so the next headings that may be level 2 headings are ignored (maybe since Doxygen expects as first a heading at level 1 corresponding to an HTML  `<H1>` tag).

To check that assumption I used the following simple README.md file
```
# A Doxygen markdown issue

This text is under the ATX heading "A Doxygen markdown issue" (level 1) that is used a title ...

## Unvisible ATX headings 

... and this text is under the ATX heading "Unvisible headings" (level 2)
that is not shown if only one level 1 heading is at top of a README.md file. 

The level of ATX headings is equal to the number of # characters in the opening sequence.
```

and doubled the first line so that after processing of the first line as  "headertitle" info the identical second line can now be interpreted as a level 1 heading (`<H1>`) and the following  level 2 heading is correctly following a level 1 heading and is so correctly processed (as `<H2>` ). 

And voila, it worked! Using a standard Doxyfile I got as browser output first
 
![image](https://user-images.githubusercontent.com/20492485/55962808-e6844a00-5c71-11e9-8cd3-e78dd56b8938.png)

and with the workaround the correct

![image](https://user-images.githubusercontent.com/20492485/55963209-b25d5900-5c72-11e9-87f6-5ed1332e9ed1.png)

I tested current doxygen release 1.8.15 and master. 

For real and mostly larger such markdown files the problem looks more annoying.

If it's meaningful to display the first level 1 heading to have the same appearance as for e.g. on GitHub or to omit that line since it's already used as Doxygen "headertitle" is another question.

Maybe the bug is correlated to the issues #6909 and #6543.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7012

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
